lab4/FiniteAutomata.py

```diff
@@ -0,0 +1,71 @@
+import re
+
+
+class FiniteAutomata:
+
+    def __init__(self):
+        self.states = []
+        self.input_symbols = []
+        self.initial_state = []
+        self.final_states = []
+        self.transitions = {}
+        # Transitions structure
+        # {A:{'1':'B', '0':'A'},
+        #  B:{'1':'B', '0':'C'},
+        #  C:{}}
+
```

Comment:
This representation for transitions does not seem to handle the case in which we do not have a dfa. If we had, for example, A,0 = A and A,0 = B, the content for  A,0 = A would be overwritten.